src/csharp/Extensions/Microsoft.Spark.Extensions.Hyperspace.E2ETest/Index/IndexConfigTests.cs

```diff
@@ -0,0 +1,94 @@
+        {
+            var indexName = "testIndexName";
+            var indexConfig = new IndexConfig(
+                indexName,
+                new List<string> { "Id" },
+                new List<string>());
+            Assert.IsType<string>(indexConfig.IndexName);
+            Assert.IsType<List<string>>(indexConfig.IndexedColumns);
+            Assert.IsType<List<string>>(indexConfig.IncludedColumns);
+
+            Builder builder = IndexConfig.Builder();
+            Assert.IsType<Builder>(builder);
+            Assert.IsType<Builder>(builder.IndexName("indexName"));
+            Assert.IsType<Builder>(builder.IndexBy("indexed1", "indexed2"));
+            Assert.IsType<Builder>(builder.Include("included1"));
+            Assert.IsType<IndexConfig>(builder.Create());
+        }
+
+        /// <summary>
+        /// Test creating an IndexConfig using its class constructor.
+        /// </summary>
+        [SkipIfSparkVersionIsLessThan(Versions.V2_4_0)]
+        public void TestIndexConfigConstructor()
+        {
+            string indexName = "indexName";
+            string[] indexedColumns = { "idx1" };
+            string[] includedColumns = { "inc1", "inc2", "inc3" };
+            var config = new IndexConfig(indexName, indexedColumns, includedColumns);
+
+            // Validate that the config was built correctly.
+            Assert.Equal(indexName, config.IndexName);
+            AssertCollectionEquals(indexedColumns, config.IndexedColumns);
+            AssertCollectionEquals(includedColumns, config.IncludedColumns);
+        }
+
+        /// <summary>
+        /// Test creating an IndexConfig using the builder pattern.
+        /// </summary>
+        [SkipIfSparkVersionIsLessThan(Versions.V2_4_0)]
+        public void TestIndexConfigBuilder()
+        {
+            string indexName = "indexName";
+            string[] indexedColumns = { "idx1" };
+            string[] includedColumns = { "inc1", "inc2", "inc3" };
+
+            Builder builder = IndexConfig.Builder();
+            builder.IndexName(indexName);
+            builder.Include(includedColumns[0], includedColumns[1], includedColumns[2]);
+            builder.IndexBy(indexedColumns[0]);
+
+            // Validate that the config was built correctly.
+            IndexConfig config = builder.Create();
+            Assert.Equal(indexName, config.IndexName);
+            AssertCollectionEquals(indexedColumns, config.IndexedColumns);
+            AssertCollectionEquals(includedColumns, config.IncludedColumns);
+        }
+
+        /// <summary>
+        /// Helper method to test that 2 collections contain the same elements.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="expected"></param>
+        /// <param name="actual"></param>
+        private void AssertCollectionEquals<T>(ICollection<T> expected, ICollection<T> actual) =>
+            Assert.True(expected.OrderBy(c => c).SequenceEqual(actual.OrderBy(c => c)));
```

Comment:
why do you order these? Can't you just call `Equals` on the `IndexConfig` object?

Reply:
This is to validate that the individual fields read correctly in the C# world.

Reply:
Removed the sorting and just assume the order is correct.